src/distance/SE2.cpp

```diff
@@ -0,0 +1,50 @@
+#include <boost/math/constants/constants.hpp>
+#include <aikido/distance/SE2.hpp>
+
+namespace aikido {
+namespace distance {
+
+//=============================================================================
+SE2::SE2(std::shared_ptr<statespace::SE2> _space)
+  : mStateSpace(std::move(_space))
+{
+  if (mStateSpace == nullptr)
+    throw std::invalid_argument("_space is nullptr.");
+}
+
+//=============================================================================
+statespace::StateSpacePtr SE2::getStateSpace() const
+{
+  return mStateSpace;
+}
+
+//=============================================================================
+double SE2::distance(
+    const aikido::statespace::StateSpace::State* _state1,
+    const aikido::statespace::StateSpace::State* _state2) const
+{
+  Eigen::VectorXd tangent1;
```

Comment:
`VectorXd` is correct here. `logMap`, which is a virtual function of `StateSpace`, takes `VectorXd` and resize it if needed.